### PR TITLE
Add `Handle::is_null()`

### PR DESCRIPTION
**Pull request by i509VCB**
This is related to a comment I made: https://github.com/ash-rs/ash/issues/573#issuecomment-1272222609

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Added `Handle::is_null` to allow checking if a handle is a null handle (#694)
```
Comment (MarijnS95) [inline Changelog.md:11]: Maybe better under `### Added`?

File: ash/src/vk.rs [hunk 2]
```diff
@@ -49,4 +49,17 @@ pub trait Handle {
     const TYPE: ObjectType;
     fn as_raw(self) -> u64;
     fn from_raw(_: u64) -> Self;
+
+    /// Returns true if the handle is a null handle.
```
Comment (MarijnS95) [inline ash/src/vk.rs:53]:
```suggestion
    /// Returns [`true`] if the handle is a null handle.
```

Or maybe just "Returns whether the handle is ..."

File: Changelog.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added `Handle::is_null` to allow checking if a handle is a null handle (#694)
 - Update Vulkan-Headers to 1.3.238 (#688)
```
Comment (MarijnS95) [inline Changelog.md:17]: I typically keeps these sorted somewhat chronologically, by PR ID ascending.

**Review by MarijnS95 — APPROVED** [review]

File: ash/src/vk.rs [hunk 4]
```diff
@@ -49,4 +49,16 @@ pub trait Handle {
...
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use ash::vk::{Handle, Instance};
+    ///
+    /// let instance = Instance::null();
+    /// assert!(instance.is_null());
+    /// ```
+    fn is_null(&self) -> bool;
```
Comment (Ralith) [inline ash/src/vk.rs:63]: Like the other methods, this should take `self` by value (see e.g. [`pointer::is_null`](https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.is_null).

**Review by Ralith — CHANGES_REQUESTED** [review]

File: ash/src/vk.rs (hunk 4 above)
Comment (MarijnS95) [reply ash/src/vk.rs]: Good catch, I did not even notice that!

File: ash/src/vk.rs [hunk 5]
```diff
@@ -49,4 +49,15 @@ pub trait Handle {
...
+    /// Returns whether the handle is a `NULL` value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use ash::vk::{Handle, Instance};
+    /// let instance = Instance::null();
+    /// assert!(instance.is_null());
+    /// ```
+    fn is_null(self) -> bool;
```
Comment (Ralith) [inline ash/src/vk.rs:62]: Maybe have a provided impl of `self.as_raw() == 0`?

**Review by Ralith — APPROVED** [review]

File: ash/src/vk.rs (hunk 5 above)
Comment (MarijnS95) [reply ash/src/vk.rs:62]: Thought about that as well, but ended up not doing that. Wouldn't hurt to have it though, I guess?

Comment (MarijnS95) [reply ash/src/vk.rs:62]: Oh apparently this is only allowed if we add `Sized` to the trait bound of `Handle`:

```rust
error[E0277]: the size for values of type `Self` cannot be known at compilation time
  --> ash/src/vk.rs:62:16
   |
62 |     fn is_null(self) -> bool {
   |                ^^^^ doesn't have a size known at compile-time
   |
help: consider further restricting `Self`
   |
62 |     fn is_null(self) -> bool where Self: std::marker::Sized {
   |                              ++++++++++++++++++++++++++++++
help: function arguments must have a statically known size, borrowed types always have a known size
   |
62 |     fn is_null(&self) -> bool {
   |                +
```

Which is fine.

Comment (Ralith) [reply ash/src/vk.rs]: The main benefit of a provided impl is that now we can drop the explicit impls from the macro.

Comment (MarijnS95) [reply ash/src/vk.rs]: D'oh, I forgot to save.

**Review by Ralith — APPROVED** [review]

**Pull request merged**